packages/json/src/lib/JSONProvider.ts

```diff
@@ -686,8 +687,12 @@ export class JSONProvider<StoredValue = unknown> extends JoshProvider<StoredValu
 
   protected async fetchVersion(context: JoshProvider.Context) {
     const { name } = context;
-    const { dataDirectoryName } = this.options;
-    const index = new ChunkIndexFile({ directory: resolve(process.cwd(), dataDirectoryName ?? 'data', name), version: this.version });
+    const { useAbsolutePath, dataDirectory } = this.options;
+    const index = new ChunkIndexFile({
+      directory: useAbsolutePath ? resolve(dataDirectory ?? 'data', name) : resolve(process.cwd(), dataDirectory ?? 'data', name),
```

Comment:
Could this be just passed to the chunkhandler instead of being determined in both files as it could lead to issues

Reply:
I'd have to look into it, but probably won't be possible just how the classes are designed.

Reply:
Aight then that's fine